src/test/java/teammates/test/cases/webapi/UpdateInstructorPrivilegeActionTest.java

```diff
@@ -0,0 +1,309 @@
+
+        InstructorPrivilegeData response = (InstructorPrivilegeData) result.getOutput();
+        assertTrue(response.isCanModifyCourse());
+        assertTrue(response.isCanModifySession());
+        assertTrue(response.isCanModifyStudent());
+        assertTrue(response.isCanModifyInstructor());
+
+        assertFalse(response.isCanViewStudentInSections());
+
+        assertFalse(response.isCanSubmitSessionInSections());
+        assertTrue(response.isCanViewSessionInSections());
+        assertTrue(response.isCanModifySessionCommentsInSections());
+    }
+
+    @Test
+    protected void testExecute_lastInstructorWithModifyInstructorPrivilege_shouldPreserve() {
+        InstructorAttributes instructor1OfCourse4 = typicalBundle.instructors.get("instructor1OfCourse4");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.INSTRUCTOR_ID, instructor1OfCourse4.getGoogleId(),
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse4.courseId,
+        };
+
+        InstructorPrivilegeUpdateRequest reqBody = new InstructorPrivilegeUpdateRequest();
+        reqBody.setCanModifyInstructor(false);
+
+        UpdateInstructorPrivilegeAction action = getAction(reqBody, submissionParams);
+
+        JsonResult result = getJsonResult(action);
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+
+        InstructorPrivilegeData response = (InstructorPrivilegeData) result.getOutput();
+        assertTrue(response.isCanModifyCourse());
+        assertFalse(response.isCanModifySession());
+        assertTrue(response.isCanModifyStudent());
+        assertTrue(response.isCanModifyInstructor());
+
+        assertTrue(response.isCanViewStudentInSections());
+
+        assertTrue(response.isCanSubmitSessionInSections());
+        assertTrue(response.isCanViewSessionInSections());
+        assertTrue(response.isCanModifySessionCommentsInSections());
+    }
+
+    @Test
+    protected void testExecute_withCourseIdAndSessionName_shouldFail() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.INSTRUCTOR_ID, instructor1OfCourse1.getGoogleId(),
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
+        };
+
+        InstructorPrivilegeUpdateRequest reqBody = new InstructorPrivilegeUpdateRequest();
+        reqBody.setCanViewStudentInSections(true);
+        reqBody.setCanSubmitSessionInSections(true);
+        reqBody.setFeedbackSessionName("session1");
+
+        assertThrows(InvalidHttpRequestBodyException.class, () -> {
+            UpdateInstructorPrivilegeAction illegalAction = getAction(reqBody, submissionParams);
+            getJsonResult(illegalAction);
+        });
+    }
+
+    @Test
+    protected void testExecute_withCourseIdAndSectionNameAndCourseLevelPrivilege_shouldFail() {
```

Comment:
Same for this.

Reply:
this is because with section name, one can only update section and session level privileges, the granularity does not match here.